File: Idle.java
import akka.actor.AbstractActor;
import akka.actor.ActorSystem;
import akka.actor.Props;
import java.util.concurrent.CountDownLatch;

public class Idle {
    
    static class IdleActor extends AbstractActor {
        private final CountDownLatch latch;

        public IdleActor(CountDownLatch latch) {
            this.latch = latch;
        }

        @Override
        public void preStart() {
            // Signal that this actor is fully initialized and ready to idle
            latch.countDown();
        }

        @Override
        public Receive createReceive() {
            return receiveBuilder().build();
        }

        public static Props props(CountDownLatch latch) {
            return Props.create(IdleActor.class, () -> new IdleActor(latch));
        }
    }

    public static void main(String[] args) throws InterruptedException {
        int nActors = 100_000;
        int durationSeconds = 30; 
        
        ActorSystem system = ActorSystem.create("IdleSystem");
        CountDownLatch latch = new CountDownLatch(nActors);

        for (int i = 0; i < nActors; i++) {
            system.actorOf(IdleActor.props(latch));
        }

        // WAIT here until all 100,000 actors are fully booted and resting
        latch.await();

        // NOW the system is truly idle. Start the measurement.
        System.out.println("LOG_START:" + System.currentTimeMillis());
        
        Thread.sleep(durationSeconds * 1000);

        System.out.println("LOG_END:" + System.currentTimeMillis());

        system.terminate();
    }
}

File: Idle.exs
defmodule Idle do
  def run do
    n_actors = 100_000
    duration_seconds = 30
    parent = self()

    # Spawn actors and have them signal when they are fully booted
    Enum.each(1..n_actors, fn _ ->
      spawn(fn ->
        send(parent, :ready)
        receive do :stop -> :ok end
      end)
    end)

    # Wait until all 100,000 processes are alive and resting
    Enum.each(1..n_actors, fn _ ->
      receive do :ready -> :ok end
    end)

    # NOW the system is truly idle. Start the measurement.
    IO.puts("LOG_START:#{System.system_time(:millisecond)}")

    # Sleep for 30 seconds
    Process.sleep(duration_seconds * 1000)

    IO.puts("LOG_END:#{System.system_time(:millisecond)}")
  end
end

Idle.run()
